### ISSUE TITLE
[Feature Request] Ely.by skin support

### ISSUE THREAD

**Issue by joex92** [SkinsRestorer/SkinsRestorerX#620]
Hi, I used to use a lot this plugin, but then found Ely.by and found it better (for newcomers mc users) because you don't need to use a link or a specific mc online username but just go to Ely.by page and change the skin there... 
I noticed Ely.by offers an [API](https://docs.ely.by/en/index.html), so I wanted to know if it could be possible to add Ely.by support to the plugin, because I tried using both plugins at the same time, but sometimes they get confused, and at the same time I like SkinRestorer better for the features than Elyby plugin (Elyby plugin works as is but it doesn't have any features like changing skin in-game for example)....

**Comment by xknat** [comment]
we will likely use another website then el.by since el.by is run by tlauncher afaik... and they are ignoring us

**Comment by imDaniX** [comment]
> we will likely use another website then el.by since el.by is run by tlauncher afaik... and they are ignoring us

Ely.by is actually a standalone project. TL Legacy (the original TLauncher) has an option to turn this system on, but the TLauncher is using its own and is selling HD skins and capes. TL;DR, TLauncher guys created a fake company and took down the original TL, they also took down some random launcher for Terraria with the same name.

**Comment by xknat** [comment]
Hmmm, will look into it,
It fly depends on how their layout is to be honest

**Comment by joex92** [comment]
yeah, Ely.by is standalone, they have their own skin plugin: https://ely.by/server-skins-system
I believe there was 1 other plugin using their api, but I don't quite remember... 
anyways, the Ely.by api gets the skins from mojang first (if they user is premium) and then it fallbacks to ely.by, but it needs a `serverKey` to be able to get the skins from ely.by, and it's easy to get one, the server owner has to register the server on the [ely.by website](https://ely.by/server-skins-system) and a key is given, then it's added to the ely.by config file and it works... so for implementing the ely.by system you might have to add an ely.by config section on the skinrestorer config file, maybe with an option for server owners to enable it or not... 
here are the docs for the Ely.by API: https://docs.ely.by/en/index.html

**Comment by xknat** [comment]
`http://skinsystem.ely.by/textures/signed/{USERNAME}?unsigned=false` to get mojang signed skin info.
Source: https://docs.ely.by/en/skins-system.html#url

This can be done since we now use the single request, so a override is "possible"

**Comment by joex92** [comment]
so is it going to be implemented? or is it already in the last release?

**Comment by xknat** [comment]
undecided, but it's possible.
Just thinking about making it a config option of fully customizable api string

**Comment by joex92** [comment]
Yeah, you could add an API option and put yours (the one you usually use) as default and commented out the ely.by one as an example how to add an API...

**Comment by joex92** [comment]
at least could you add a workaround for this:
the ely.by API can get the skin file with just the username, e.g.,  like this: http://skinsystem.ely.by/skins/JoeX92
but it will redirect it to the image file resolving in this link: http://ely.by/minecraft/skins/6dc7150a8ac5a4d3e882e3b6a518b3c0.png
using it with the `/skin set` command resolves in an error
![image](https://user-images.githubusercontent.com/645579/145288711-079e214c-0fcd-4ccb-9ccf-9e26fd932805.png)

**Comment by xknat** [comment]
> at least could you add a workaround for this: the ely.by API can get the skin file with just the username, e.g., like this: http://skinsystem.ely.by/skins/JoeX92 but it will redirect it to the image file resolving in this link: http://ely.by/minecraft/skins/6dc7150a8ac5a4d3e882e3b6a518b3c0.png using it with the `/skin set` command resolves in an error ![image](https://user-images.githubusercontent.com/645579/145288711-079e214c-0fcd-4ccb-9ccf-9e26fd932805.png)

would work if the skin is valid, sadly tlauncher does allow illegal skin format / options which mojang and the real game do not support. for the "fake skin / HD" to work you need everyone to use a mod.

**Comment by joex92** [comment]
._. what I need is that for the people that haven't bought the game, they could still have a skin by using ely.by system without having to download any mod... Ely.By offers an API... the skin provided is valid, look at this:
![image](https://user-images.githubusercontent.com/645579/145308319-d5c655f9-9ae9-42e7-b683-b6441e96752c.png)
![image](https://user-images.githubusercontent.com/645579/145308533-4780de54-9356-4a7c-a5f3-a04a11c7fda7.png)
I used the redirected link instead of the API link with the username: http://skinsystem.ely.by/skins/JoeX92 
but when I use the API link with the username, it says error because the plugin doesn't resolve the link, it just asumes that the given link is the image itself... that's what I'm talking about...

**Comment by xknat** [comment]
> ._. what I need is that for the people that haven't bought the game, they could still have a skin by using ely.by system without having to download any mod... Ely.By offers an API... the skin provided is valid, look at this: ![image](https://user-images.githubusercontent.com/645579/145308319-d5c655f9-9ae9-42e7-b683-b6441e96752c.png) ![image](https://user-images.githubusercontent.com/645579/145308533-4780de54-9356-4a7c-a5f3-a04a11c7fda7.png) I used the redirected link instead of the API link with the username: http://skinsystem.ely.by/skins/JoeX92 but when I use the API link with the username, it says error because the plugin doesn't resolve the link, it just asumes that the given link is the image itself... that's what I'm talking about...

No its NOT a invalid skin
The skin is not conform https://minecraft.fandom.com/wiki/Skin 
The skins that the mod used do not have to be skins that the game it self can show. That's your misunderstanding and because of that you think that those "hd skins" can be set in the game which is not true as we must have any skin signed by mojang and mojang will not sign those skins as they are invalid.
